src/lib/Components/Consignments/Screens/Overview.tsx

```diff
@@ -56,18 +55,31 @@ export default class Info extends React.Component<Props, ConsignmentSetup> {
   goToLocationTapped = () =>
     this.props.navigator.push({ component: Location, passProps: { updateWithResult: this.updateLocation } })
 
-  updateMetadata = (result: ConsignmentMetadata) => this.setState({ metadata: result })
-  updateProvenance = (result: string) => this.setState({ provenance: result })
+  updateArtist = (result: SearchResult) => this.updateStateAndMetaphysics({ artist: result })
+  updateMetadata = (result: ConsignmentMetadata) => this.updateStateAndMetaphysics({ metadata: result })
+  updateProvenance = (result: string) => this.updateStateAndMetaphysics({ provenance: result })
   updateLocation = (city: string, state: string, country: string) =>
-    this.setState({ location: { city, state, country } })
+    this.updateStateAndMetaphysics({ location: { city, state, country } })
+
+  updateStateAndMetaphysics = (state: any) => this.setState(state, this.updateMetaphysics)
 
-  createSubmission = () =>
+  updateMetaphysics = async () => {
+    if (this.state.submission_id) {
+      updateSubmission(this.state, this.state.submission_id)
+    } else if (this.state.artist) {
+      const submission = await createSubmission(this.state)
+      this.setState({ submission_id: submission.id })
```

Comment:
Next up I'll save this so that the next time you launch it keeps everything around https://github.com/artsy/consignments/issues/84